lib/puppet_x/templates/init_ps.ps1.erb

```diff
@@ -501,19 +498,30 @@ function Invoke-PowerShellUserCode
   }
   finally
   {
-    [Void]$event.Set()
-    [Void]$event.Close()
-    if ($PSVersionTable.CLRVersion.Major -ge 3) {
-      [Void]$event.Dispose()
-    }
+    Signal-Event -EventName $EventName
     if ($ps -ne $null) { [Void]$ps.Dispose() }
   }
 }
 
+function Signal-Event
+{
+  [CmdletBinding()]
+  param(
+    [String]
+    $EventName
+  )
+
+  $event = [System.Threading.EventWaitHandle]::OpenExisting($EventName)
+
+  [System.Diagnostics.Debug]::WriteLine("Signaling event $EventName")
```

Comment:
I'm not really fussed either way.  If we keep it in, I would expect that we would properly instrument this with debug statements later.